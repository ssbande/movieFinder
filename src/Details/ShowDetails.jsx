import React from 'react';
import PropTypes from 'prop-types';
import { Grid, Header, Image } from 'semantic-ui-react';

import { createImageUrl } from '../utils/helpers';

const ShowDetail = ({ detailInfo }) => {
  const { poster_path, name, overview, genres } = detailInfo;

  return <Grid stackable stretched>
    <Grid.Column width={5}>
      <Image className="posterImage" src={createImageUrl(poster_path)} size='medium' bordered />
    </Grid.Column>
    <Grid.Column width={11}>
      <div>
        <Header size='huge' color='yellow' inverted dividing>{name}</Header>
        <div>{overview}</div>
        <Header as='h3' inverted color='grey'>
          Genres
        <Header.Subheader>{genres.map(genre => (genre.name)).join(', ')}</Header.Subheader>
        </Header>
      </div>
    </Grid.Column>
  </Grid>
}

ShowDetail.propTypes = {
  detailInfo: PropTypes.shape({
    poster_path: PropTypes.string,
    name: PropTypes.string,
    overview: PropTypes.string,
    genres: PropTypes.arrayOf(PropTypes.shape({
      name: PropTypes.string
    }))
  })
}

export default ShowDetail;