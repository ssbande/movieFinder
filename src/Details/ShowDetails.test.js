import React from 'react';
import { shallow } from 'enzyme';

import ShowDetails from './ShowDetails';

const setup = () => ({
  props: {
    detailInfo: {
      poster_path: 'poster_path',
      name: 'name',
      overview: 'overview',
      genres: [{name: 'genre1'}, {name: 'genre2'}, {name: 'genre3'}]
    }
  }
})

describe('Movie Detail', () => {
  it('should render correctly', () => {
    const { props } = setup();
    const wrapper = shallow(<ShowDetails {...props} />);
    expect(wrapper).toMatchSnapshot();
  })
})