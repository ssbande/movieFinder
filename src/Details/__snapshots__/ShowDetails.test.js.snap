// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie Detail should render correctly 1`] = `ShallowWrapper {}`;
